.github/workflows/python.yml

```diff
@@ -10,19 +10,18 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v4
-    # NB: v1.4.0 covers Python 3.8.
-    - uses: ricardochaves/python-lint@v1.4.0
+    - uses: actions/setup-python@v5
       with:
-        python-root-list: lddtree.py pylint
-        use-pylint: true
-        use-pycodestyle: false
-        use-flake8: false
-        use-black: true
-        use-mypy: true
-        use-isort: true
-        extra-pylint-options: ""
-        extra-pycodestyle-options: ""
-        extra-flake8-options: ""
-        extra-black-options: ""
-        extra-mypy-options: ""
-        extra-isort-options: ""
+        python-version: '3.x'
+    - name: install dependencies
+      run: |
+        python -m pip install pylint mypy black 'isort[colors]'
+        # mypy dependencies for following imported types
+        python -m pip install argcomplete
```

Comment:
do we need to run pip twice ?  seems better to merge them

Reply:
It's not meaningfully slower, afaik, and using two pip lines means you can comment one of them separately from the rest. There's no deeper reason.

Reply:
it's bash, so you could use an array to inline things, and to split "define list of packages" out from "use list of packages"

```
pkgs=(
  # The tools we care about.
  ...
  # Extra 3rd party deps to enable linting/typing/etc...
  ...
)
python -m pip install "${pkgs[@]}"
```